src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -229,6 +230,19 @@ private async ValueTask<CancellationTokenRegistration> HandleWriteStartState(Can
                 throw new InvalidOperationException(SR.net_quic_writing_notallowed);
             }
 
+            if (cancellationToken.IsCancellationRequested)
+            {
+                lock (_state)
+                {
+                    if (_state.SendState == SendState.None || _state.SendState == SendState.Pending)
+                    {
+                        _state.SendState = SendState.Aborted;
+                    }
+                }
```

Comment:
It's up to you, but most streams that do upfront checks don't manipulate the state of the stream as a result.  This would simply become:
```C#
cancellationToken.ThrowIfCancellationRequested();
```

Reply:
I did that originally (without cancellationToken.ThrowIfCancellationRequested)
I end up failing this test:

https://github.com/dotnet/runtime/blob/183c4d100f68fb6c177a1fe71809d581aa25e47b/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs#L468-L483

It would cancel `WriteAsync` _with_ cancellation token but than it expects that next `WriteAsync` _without_ cancellation would be cancelled as well. This is specific to QUIC, not the conformance suite. 

I'm not sure if that is what we want but that was reason to add the state transition as well e.g. make the stream unusable after _any_ cancellation. In this case, the first really does not touch MsQuic and does not (need to) make any state changes so the subsequent WriteAsync could work without any issues. But then it would be less consistent and perhaps harder to explain.

Reply:
If that is desirable, it's ok.  Most other streams don't do that, so it's arguably the test being too constrained.  But in reality I don't think it hurts.